ansible/playbooks/roles/repository/tasks/setup.yml

```diff
@@ -9,35 +9,50 @@
   debug:
     var: offline_mode
 
-- name: Set directory name for download script
+- name: Set OS name for download script
   set_fact:
-    download_script_subdir: >-
+    download_requirements_os_name: >-
       {{ 'centos-7' if (ansible_distribution == 'CentOS' and ansible_distribution_major_version == '7') else
           'redhat-7' if (ansible_distribution == 'RedHat' and ansible_distribution_major_version == '7') else
           'ubuntu-20.04' if (ansible_distribution == 'Ubuntu' and ansible_distribution_version == '20.04') else None }}
 
-- name: Download Epiphany requirements on repository host # online mode
+- name: Set paths for download-requirements script
+  set_fact:
+    download_requirements_dir: "{{ _download_requirements_dir }}"
+    download_requirements_script: "{{ _download_requirements_dir }}/download-requirements.py"
+    download_requirements_flag: "{{ _download_requirements_dir }}/download-requirements-done.flag"
+  vars:
+    _download_requirements_dir: "/var/tmp/epi-download-requirements"
+
+- name: Setup venv and download Epiphany requirements on repository host  # online mode
   block:
-    - name: Check if OS distribution is supported
-      assert:
-        that: download_script_subdir | length > 0
-        fail_msg: "Your OS distribution is not supported"
+    - name: Install required packages for repository
+      include_tasks: "{{ ansible_os_family }}/install-packages.yml"
+
+    - name: Install Poyo library for YAML support
+      pip:
+        name: poyo==0.5.0
+        executable: pip3
+
+    - name: Create download-requirements directory
+      file:
+        path: "{{ download_requirements_dir }}"
+        state: directory
 
     - name: Copy files for downloading requirements to repository host
-      copy:
-        src: download-requirements/{{ download_script_subdir }}/
-        dest: /var/tmp/epi-download-requirements
+      synchronize:
+        src: download-requirements/
```

Comment:
It looks like this will copy all files regardless of distribution and architecture. For example if we have Ubuntu there is no need to copy RHEL specific files. IMO to be improved in the future, please create an issue for this.

Reply:
Created an issue for that: [2975](https://app.zenhub.com/workspaces/epiphany-platform-5ef9d3c749afc20019782391/issues/epiphany-platform/epiphany/2975)